packages/embark-ui/src/containers/ContractsContainer.js

```diff
@@ -23,14 +29,58 @@ class ContractsContainer extends Component {
     this.props.stopContracts();
   }
 
+  get numberOfContracts() {
+    return this.props.contracts
+      .filter(contract => !contract.silent).length;
+  }
+
+  getNumberOfPages() {
+    return Math.ceil(this.numberOfContracts / this.numContractsToDisplay);
+  }
+
+  changePage(newPage) {
+    this.setState({currentPage: newPage});
+    this.props.fetchContracts();
+  }
+
+  getCurrentContracts() {
+    let offset = 0;
+    return this.props.contracts
+      .filter((contract, arrIndex) => {
+        if (contract.silent) {
+          offset++;
+          return false
+        };
+        const index = (
+          (arrIndex + 1 - offset) -
+            (this.numContractsToDisplay * (this.state.currentPage - 1))
+        );
+        return index <= this.numContractsToDisplay && index > 0;
+      });
+  }
+
   render() {
+    const newContracts = this.getCurrentContracts();
+    let ContractsComp;
+    if (this.props.mode === "detail") {
+      ContractsComp = Contracts
+    } else if (this.props.mode === "list") {
+      ContractsComp = ContractsList
+    }
     return (
       <React.Fragment>
-        {this.props.updatePageHeader && <PageHead title="Contracts" description="Summary of all deployed contracts" />}
-        <DataWrapper shouldRender={this.props.contracts.length > 0} {...this.props} render={({contracts}) => {
-            if (this.props.mode === "list") return <ContractsList contracts={contracts} />;
-            if (this.props.mode === "detail") return <Contracts contracts={contracts} />;
-        }} />
+        {this.props.updatePageHeader &&
+         <PageHead title="Contracts"
+                   description="Summary of all deployed contracts" />}
+        <DataWrapper
+          shouldRender={true}
```

Comment:
You don't need the DataWrapper if you always render it.
You can then just put the `ContractsComp`

Reply:
Cool.  The containers I worked on all had it, and I hadn't studied it so assumed it needed to be there.

Reply:
Done. See WIP commit.